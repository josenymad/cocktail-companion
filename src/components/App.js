import React from "react";
import { Routes, Route } from "react-router-dom";
import "../styles/App.css";
import NavBar from "./NavBar";
import BestBarware from "./Barware";

const App = () => {
  return (
    <div className="App">
      <NavBar />
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/best-barware" element={<BestBarware />} />
        <Route path="all-cocktails" element={<AllCocktails />} />
      </Routes>
    </div>
  );
};

export default App;
