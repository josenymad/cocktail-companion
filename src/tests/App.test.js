import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import axios from "axios";
import App from "../components/App";

describe("App", () => {
  it("renders correctly", async () => {
    const mockData = {
      drinks: [
        {
          idDrink: "1",
          image: "mock-url1",
          title: "mock-title1",
        },
      ],
    };
    jest.spyOn(axios, "get").mockResolvedValue(mockData);
    const { asFragment } = await render(
      <MemoryRouter>
        <App />
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
