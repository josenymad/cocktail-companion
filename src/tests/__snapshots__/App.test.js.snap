// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <div
      class="navbar"
    >
      <div
        class="navbar-titles"
      >
        <h1
          class="navbar-title__header"
        >
          Cocktail Companion
        </h1>
        <h2
          class="navbar-title__slogan"
        >
          The Ultimate Cocktail Guide
        </h2>
      </div>
      <ul
        class="navbar-list"
      >
        <li
          class="navbar-list__item"
        >
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="navbar-list__item"
        >
          <a
            href="/best-barware"
          >
            Best Barware
          </a>
        </li>
        <li
          class="navbar-list__item"
        >
          <a
            href="/all-cocktails"
          >
            All Cocktails
          </a>
        </li>
        <li
          class="navbar-list__item"
        >
          <form>
            <label
              for="search"
            />
            <input
              id="search"
              type="search"
              value=""
            />
            <label
              for="search-button"
            />
            <button
              class="navbar-list__button"
              id="search-button"
              type="submit"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-magnifying-glass "
                data-icon="magnifying-glass"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </form>
        </li>
      </ul>
    </div>
    <div
      class="home"
    >
      <p
        class="home__text"
      >
        Welcome to Cocktail Companion! We have got you covered with an extensive range of delicious drinks, barware to suit all budgets and easy to follow instructions to get you mixing cocktails in no time. Search by ingredient, check out our favourites or let us surprise you with our random cocktail generator!
      </p>
    </div>
  </div>
</DocumentFragment>
`;
